jupyter_core/version.py

```diff
@@ -1,6 +1,24 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-version_info = (4, 6, 4, "dev")
+from collections import namedtuple
 
-__version__ = '.'.join(map(str, version_info))
+VersionInfo = namedtuple('VersionInfo', [
+    'major',
+    'minor',
+    'micro',
+    'releaselevel',
+    'serial'
+])
+
+version_info = VersionInfo(4, 6, 4, 'alpha', 0)
+
+_specifier_ = {'alpha': 'a', 'beta': 'b', 'candidate': 'rc', 'final': ''}
```

Comment:
Is dropping the developmental specifier intentional?  I view all these as being tied to an actual release or pre-release posted to pypi while `dev` is something I know is locally built.

I'd be in favor of adding a `'development': 'dev'` or `'dev': 'dev'` entry, but if that's not the convention then I suppose that's okay, although a number of Jupyter repos still use 'dev'.

Reply:
Sure. We were going off of the python conventions in the python source (https://github.com/python/cpython/blob/4b9aad49992a825d8c76e428ed1aca81dd3878b2/Python/sysmodule.c#L2534-L2541), but I see that pep 440 also has development releases as an official convention: https://www.python.org/dev/peps/pep-0440/#developmental-releases

I'll add a `'development': 'dev'` entry too.

Reply:
Ah, yeah, I figured you were using PEP 440.  Thanks.  Should the suffix be '.dev' in this case - since that's the current convention?

Reply:
Yep, I fixed that too in the commit I just pushed.